lib/determine-basal/determine-basal.js

```diff
@@ -979,12 +979,19 @@ var determine_basal = function determine_basal(glucose_status, currenttemp, iob_
             } else {
                 rT.reason += "Eventual BG " + convert_bg(eventualBG, profile) + " > " + convert_bg(min_bg, profile) + " but Min. Delta " + minDelta.toFixed(2) + " < Exp. Delta " + convert_bg(expectedDelta, profile);
             }
-            if (currenttemp.duration > 15 && (round_basal(basal, profile) === round_basal(currenttemp.rate, profile))) {
-                rT.reason += ", temp " + currenttemp.rate + " ~ req " + basal + "U/hr. ";
-                return rT;
+
+            // don't increase the basal rate in response to a low delta
+            if (currenttemp.duration > 0 && basal > currenttemp.rate) {
+                rT.reason += "; extending current temp basal of " + currenttemp.rate + " U/hr. for 10 minutes. ";
+                return tempBasalFunctions.setTempBasal(currenttemp.rate, currenttemp.duration + 10, profile, rT, currenttemp);
```

Comment:
This won’t work as you’d expect: the pump requires all temp basals to be set at multiples of 30m.